### ISSUE TITLE
_get_per_token_logps_and_entropies in GRPO implements the wrong logit slicing

### ISSUE THREAD

**Issue by ahatamiz** [huggingface/trl#3681]
### Reproduction

Simply as the title suggests, [_get_per_token_logps_and_entropies](https://github.com/huggingface/trl/blob/e04f7eb3b993b5278733a793853ed3a85d1c69c8/trl/trainer/grpo_trainer.py#L867) incorrectly assumes that the sequence length of the `logits` tensor should just be reduced by one. However, what's needed is to select only the `logits` that correspond to the `completion_ids`. Since the `logits` for the entire `prompt + completion` sequence are returned, we need to slice out only the part that aligns with the completions.

### System Info

```
vllm=0.9.1
trl[vllm]=0.19.0
transformers=4.52.3
torch=2.6.0
```

### Checklist

- [x] I have checked that my issue isn't already filed (see [open issues](https://github.com/huggingface/trl/issues?q=is%3Aissue))
- [x] I have included my system information
- [x] Any code provided is minimal, complete, and reproducible ([more on MREs](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks))
- [x] Any code provided is properly formatted in code blocks, (no screenshot, [more on code blocks](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks))
- [x] Any traceback provided is complete
